### PR TITLE
32 write the code to register a new user in the server

### PR DESCRIPTION
**Pull request by MariaAguiar**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,7 +6,7 @@ NAME=		ircserv
 
 ### COMPILATION ###
 CC=			c++
-CFLAGS=		-Wall -Werror -Wextra -g -std=c++98 -fsanitize=address
+CFLAGS=		-Wall -Werror -Wextra -g -std=c++98 #-fsanitize=address
```
Comment (JonasBuchadas) [inline Makefile:9]: Is this flag disable for any reason ?

File: incl/User.hpp [hunk 2]
```diff
@@ -6,14 +6,15 @@
...
+  User &operator=( User const &src );
 
  public:
-  User();
+  User(std::string serverName, std::string nickName);
   ~User();
-  User( User const &src );
-  User &operator=( User const &src );
+  
+  std::string _serverName;
```
Comment (JonasBuchadas) [inline incl/User.hpp:16]: Is there any reason to not use the attributes as private and have getters and setters ?

**Review by JonasBuchadas — COMMENTED** [review]
Just to confirm. Are you using a different formatter ? Many of these changes are style related.

File: Makefile (hunk 1 above)
Comment (MariaAguiar) [reply Makefile:9]: I use valgrind more. It shows uninitialized var problems. Can be put again

File: incl/User.hpp (hunk 2 above)
Comment (MariaAguiar) [reply incl/User.hpp:16]: You're right. Will do.

File: Makefile (hunk 1 above)
Comment (JonasBuchadas) [reply Makefile:9]: Perfect!

**Review by JonasBuchadas — APPROVED** [review]

**Pull request merged**